packages/cdktf-cli/bin/cmds/ui/watch.tsx

```diff
@@ -0,0 +1,271 @@
+import React, { useEffect, useMemo, useState } from "react";
+import { Text, Box, Static, Spacer } from "ink";
+import { Props as TextProps } from "ink/build/components/Text";
+import Spinner from "ink-spinner";
+import stripAnsi from "strip-ansi";
+import useStdoutDimensions from "ink-use-stdout-dimensions";
+import { gql, useSubscription } from "@apollo/client";
+import { GraphQLWatchState } from "../../../lib/server/util";
+import format from "date-fns/format";
+
+const StatusBox = ({
+  status,
+  stackName,
+  error,
+}: {
+  status: string;
+  stackName?: string;
+  error?: GraphQLWatchState["error"];
+}) => {
+  const [didDeployAlready, setDidDeployAlready] = useState(false);
+  useEffect(() => {
+    if (status === "DEPLOYING" && !didDeployAlready) setDidDeployAlready(true);
+  }, [didDeployAlready, status]);
+
+  let statusText = <Text>{status}</Text>;
+  const stack = stackName ? <Text color="blueBright">{stackName}</Text> : null;
+  switch (status) {
```

Comment:
Nitpick: moving this into a function would make the component more readable IMHO :)

Reply:
You mean multiple different components instead of this big switch?

Reply:
Going to keep it this way for now.